File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int>v;
        v.push_back(nums[0]);
        for(int i=1;i<nums.size();++i){
            if(nums[i]>v.back()){
                v.push_back(nums[i]);
            }else{
                int idx = lower_bound(begin(v),end(v),nums[i])-v.begin();
                // cout<<idx<<endl;
                v[idx]=nums[i];
            }
        }
        return v.size();
    }
};

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [ 0 for x in range(len(nums))]
        dp[0]=1
        for x in range(1,len(nums)):
            dp[x]=1
            for y in range(x):
                if(nums[y]<nums[x]):
                    dp[x]=max(dp[x],dp[y]+1)
        return max(dp)